onnxscript/function_libs/torch_lib/graph_building.py

```diff
@@ -725,10 +756,14 @@ def to_model_proto(
         )
 
         try:
-            onnx_model = onnx.shape_inference.infer_shapes(
-                onnx_model, check_type=True, strict_mode=False, data_prop=True
-            )
-            onnx.checker.check_model(onnx_model, full_check=True)
+            if not cache_model_to_disk:
```

Comment:
In the follow up PR we will remove the checks altogether from a discussion with Aaron: we should not check it here.